File: public/examples/integrations/toolkits/search/walmart/search_walmart_products_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const TOOL_NAME = "Search.SearchWalmartProducts";

const toolInput = {
	query: "Apple iPhone",
	sort_by: "lowest_price_first",
	max_price: 250,
	next_day_delivery: true,
	page: 1,
};

const response = await client.tools.execute({
	tool_name: TOOL_NAME,
	input: toolInput,
});

console.log(response);

File: public/examples/integrations/toolkits/search/walmart/search_walmart_products_example_call_tool.py
from arcadepy import Arcade
from arcade_search.enums import WalmartSortBy

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "you@example.com"
TOOL_NAME = "Search.SearchWalmartProducts"

tool_input = {
    "query": "Apple iPhone",
    "sort_by": WalmartSortBy.PRICE_LOW_TO_HIGH,
    "max_price": 250,
    "next_day_delivery": True,
    "page": 1,
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(response)
